src/analysis/processing/qgsalgorithmfixgeometries.cpp

```diff
@@ -99,6 +112,19 @@ QgsFeatureList QgsFixGeometriesAlgorithm::processFeature( const QgsFeature &feat
     outputFeature.clearGeometry();
     return QgsFeatureList() << outputFeature;
   }
+  else
+  {
+    if ( mLogFixedErrors )
+    {
+      const Qgis::GeometryValidityFlags flags = Qgis::GeometryValidityFlags();
+      QVector<QgsGeometry::Error> errors;
+      outputFeature.geometry().validateGeometry( errors, Qgis::GeometryValidationEngine( Qgis::GeometryValidationEngine::Geos ), flags );
```

Comment:
I have a reservation here -- the GEOS validation method only ever returns the FIRST descriptive error message, not all of them. So my concern is that the report will be misleading as we're actually potentially fixing a lot of additional geometry issues vs what we report...

Reply:
Thanks for pointing that out, @nyalldawson

In that case, we could add a note to the help string to make users aware. I think getting the first error encountered per feature is still good for users that would like to have a log of fixed errors (which will be totally optional).

BTW, wouldn't this also be an issue in `qgis:checkvalidity` algorithm when selecting the `GEOS` method?

Reply:
> I think getting the first error encountered per feature is still good for users that would like to have a log of fixed errors

Hmm. I'm strongly -1 to introducing misleading results in qgis in any form -- we just don't know what the consequences of these are, or how seriously they may affect someone's analysis and results. And in this case I don't like the idea that we may misleading inform a user that eg only a duplicate vertex was removed (harmless) when in fact we also changed some intersecting inner ring in some way (with potentially serious implications).

Qgis core should have a strict no-tolerance policy for any misleading analysis results like this, and leave the feature up to a plugin where it's all "user beware".

> wouldn't this also be an issue in qgis:checkvalidity algorithm when selecting the GEOS method?

Yes. That's why we have the qgis validator and why it's used by default instead of the geos one while editing features. (Also why you have to iteratively run the validate algorithm with geos validator after manually fixing each reported error)

Reply:
@nyalldawson please don't get me wrong, of course I'd like to make QGIS better and won't suggest to add anything that makes QGIS worst at any level. I won't spend my time on that.

Perhaps I'm missing something, but if I see the option GEOS in the qgis:checkvalidity alg, I think it's there because it'll give users a whole response on the geometry validation. Based on that assumption, I wanted to move those results to the fixgeometries algorithm as an optional output.

With the context you've given, I agree, there is no point in adding the new option to fixgeometries. However, I'd still suggest to add a small note to qgis:checkvalidity's help, so that users (and other devs) can be aware of an incomplete result from the geometry analysis (i.e., only the first error encountered) with the GEOS mode.

Would you be in favor of converting this PR in a PR that adds such a note to qgis:checkvalidity's help?

Reply:
>  However, I'd still suggest to add a small note to qgis:checkvalidity's help, so that users (and other devs) can be aware of an incomplete result from the geometry analysis (i.e., only the first error encountered) with the GEOS mode.

> Would you be in favor of converting this PR in a PR that adds such a note to qgis:checkvalidity's help?

Yes, definetely!

If you want more to give more context, then something like "The GEOS method is faster and performs better on larger geometries, but is limited to only returning the first error encountered in a geometry. The QGIS method will be slower but reports all errors encountered in the geometry, not just the first."